extpsdk/README.md

```diff
@@ -0,0 +1,260 @@
+## Repository Contents
+
+This repository contains the SDK for building Vantiq connectors (also known as extension sources)
+using Python. At a high level, a connector is configured and then run, responding or sending messages to messages the Vantiq server. Each connector acts as a _source_ in a Vantiq namespace.
+
+You will need credentials to a Vantiq server to use this SDK.  Please contact your Vantiq system administrator to obtain these credentials.
+
+## Overview of a Connector
+
+For details about the overall structure of a connector and the protocol used between the connector and the Vantiq source, please see [here](../README.md).  
+
+Each running instance of this code represents a set of connectors that connect to a single Vantiq namespace. The set of of sources for the connector & the connection information is found in the server config file.  We will describe the details later.
+
+Using the Python VantiqConnector SDK, the overall operation is as follows:
+
+* Create a `VantiqConnectorSet` that manages the set of connections (one per source listed in the `server.config` file.
+* Establish a set of _handlers_ or callbacks that will allow the connectors to respond to messages from the Vantiq server. There are callbacks for the following situations:
+    * `connect` -- when your connection to the server is established, the source configuration is provided to the connector. This can be used to tailor the behavior or local work.
+    * `publish` -- when the Vantiq server sends an event to the source, that event is delivered to the `publish` handler
+    * `query` -- when the Vantiq server queries the source, that query is delivered to the `query` handler.  The query handler can return information or errors to the application running on the server.
+    * `close` -- when the connection to the Vantiq server ends, the `close` handler is called, giving the connector a chance to clean up resources.
+
+The connector runtime will maintain the connection, reconnecting as needed.
+
+The following sections describe the use of this SDK to construct a connector.
+
+## Building a Connector
+
+Running this code required Python 3.8 or later.
+
+To use the SDK, install it into your Python environment.  Using `pip`, 
+
+```commandline
+pip install vantiqconnectorsdk
+```
+
+Other Python environment tools may use different commands.
+
+Once it is installed, you can import it.
+
+```python
+
+from vantiqconnectorsdk import VantiqConnector, VantiqConnectorSet, VantiqConnectorException, VantiqSourceConnection
+```
+
+The SDK is built using _asyncio_.  For details about working with _asyncio_, please see [Python `asyncio` documentation](https://docs.python.org/3/library/asyncio.html).
+
+The Vantiq Connector SDK includes the following items:
+
+* VantiqConnector -- a set of conststants defined as part of the interface.
```

Comment:
```suggestion
* VantiqConnector -- a set of constants defined as part of the interface.
```

Might also be clearer if you move this a few lines down to where the constants are listed.